Microsoft.Toolkit.Parsers/Markdown/Helpers/Common.cs

```diff
@@ -511,5 +511,32 @@ public static int SkipQuoteCharacters(string markdown, int start, int end, int q
 
             return startOfLine;
         }
+
+        /// <summary>
+        /// Checks if the given URL is allowed in a markdown link.
+        /// </summary>
+        /// <param name="url"> The URL to check. </param>
+        /// <returns> <c>true</c> if the URL is valid; <c>false</c> otherwise. </returns>
+        public static bool IsUrlValid(string url)
+        {
+            // URLs can be relative.
+            if (!Uri.TryCreate(url, UriKind.Absolute, out Uri result))
+            {
+                return true;
+            }
+
+            // Check the scheme is allowed.
+            bool schemeIsAllowed = false;
```

Comment:
could probably do without this flag .. just return true immediately in the loop .. and return false after the loop